Chat/FileManager.cpp

```diff
@@ -0,0 +1,96 @@
+#include "FileManager.h"
+
+FileManager::FileManager()
+{
+    Load();
+}
+
+FileManager::~FileManager()
+{
+    Save();
+}
+
+bool FileManager::AddFile(const std::string &path)
+{
+    File new_file;
+    if(!new_file.SetFile(path))
+    {
+        return false;
+    }
+
+    shared_files_.push_back(new_file);
+    return true;
+}
+
+void FileManager::RemoveFile(int file_index)
+{
+    shared_files_.erase(shared_files_.begin() + file_index);
+}
+
+std::string FileManager::GetFilePath(int file_index)const
+{
+    return shared_files_[file_index].GetPath();
+}
+
+std::string FileManager::GetFileName(int file_index)const
+{
+    return shared_files_[file_index].GetName();
+}
+
+std::vector<std::string> FileManager::GetFilePaths()const
+{
+    std::vector<std::string> return_vector;
+    for(File it : shared_files_)
+    {
+        return_vector.push_back(it.GetPath());
+    }
+    return return_vector;
+}
+
+std::vector<std::string> FileManager::GetFileNames()const
+{
+    std::vector<std::string> return_vector;
+    for(File it : shared_files_)
+    {
+        return_vector.push_back(it.GetName());
+    }
+    return return_vector;
+}
+
+void FileManager::Save()
+{
+    if((CreateDirectory(FILE_DATA_SAVE_DIR, NULL)) ||
```

Comment:
Pretty good.
Using filestream is not a big deal. 
We want for you to use WINAPI not only in multythreading.
Please, use CreateFile, WriteFile, FILE functionality for work with files.